airbyte-integrations/connectors/destination-mongodb/src/main/java/io/airbyte/integrations/destination/mongodb/MongodbDestination.java

```diff
@@ -114,51 +121,43 @@ public AirbyteMessageConsumer getConsumer(JsonNode config,
 
   /* Helpers */
 
-  private MongoClientURI getMongoClientURI(JsonNode config) {
-    MongoClientURI uri = null;
-    if (config.get(AUTH_TYPE).get(AUTHORIZATION).asText().equals(LOGIN_AND_PASSWORD)) {
-      uri = new MongoClientURI(String.format(MONGO_URI, config.get(AUTH_TYPE).get(USERNAME).asText(),
-          config.get(AUTH_TYPE).get(PASSWORD).asText(), config.get(HOST).asText(),
-          config.get(PORT).asText(), config.get(DATABASE).asText()));
-    }
-    return uri;
+  private MongoDatabase getDatabase(JsonNode config) {
+    return new MongoDatabase(getConnectionString(config), config.get(DATABASE).asText());
   }
 
-  private MongoClient getMongoClient(String host, int port, @Nullable MongoClientURI uri) {
-    try {
-      if (uri == null) {
-        var serverAddress = new ServerAddress(host, port);
-        return new MongoClient(serverAddress);
+  private String getConnectionString(JsonNode config) {
```

Comment:
This method has lots of `if` branches. Can you add a unit test for it?

Reply:
Sure, I've changed method according spec changes and added a test.